BlueBrain/Search
# [BBS-279] Bump numpy version to 1.20

## Description

This PR bumps `numpy` version to `>1.20` so that [type annotations are provided out of the box](https://numpy.org/devdocs/release/1.20.0-notes.html#numpy-is-now-typed).

This means that we can drop `numpy-stubs` introduced in #227.

Fixes [BBS-279](https://bbpteam.epfl.ch/project/issues/browse/BBS-279).

## How to test this PR
Currently all `mypy` tests are running w/o error, so to check that `mypy` is doing its job you can try the following.
1. `pip uninstall numpy-stubs`
2. Replace a `np.array` by a `np.Array` anywhere you want in our source code.
3. `tox -e lint` should return the following error
```bash
src/bluesearch/embedding_models.py:577: error: Module has no attribute "Array"; maybe "array"?  [attr-defined]
```

## Checklist:

- [x] This PR refers to an issue present in the [issue tracker](https://github.com/BlueBrain/Search/issues
) (if it is not the case, please create an issue
 first).
- [x] Travis CI pipeline run (add `[test travis]` to your git commit to run it)
